Sources/Fuzzilli/FuzzIL/JsOperations.swift

```diff
@@ -1047,21 +1047,37 @@ final class TestIn: JsOperation {
 
 }
 
+enum ParameterPattern {
+    case identifier
+    case object(properties: [ObjectPatternProperty])
+    case array(elements: [ParameterPattern])
+}
+
+struct ObjectPatternProperty {
+    let key: String
+    let value: ParameterPattern
+}
+
 // The parameters of a FuzzIL subroutine.
 public struct Parameters {
-    /// The total number of parameters.
     private let numParameters: UInt32
-    /// Whether the last parameter is a rest parameter.
     let hasRestParameter: Bool
```

Comment:
Maybe leave a TODO here to investigate if we want to have this also be a pattern